### PR TITLE
Support typescript 5.2

### LINKED ISSUE #40: Support Typescript 5.2

### PR DESCRIPTION
**Pull request by carlansley**
Fixes #40.

Now a major update, because assumes publishing is done with `builder`.  the good news is that `tsconfig.json` is now just:
```
{
  "extends": "@checkdigit/typescript-config"
}
```

### CODE REVIEW

**Comment by carlansley** [comment]
Waiting on https://github.com/typescript-eslint/typescript-eslint/issues/7155 to get rid of lint warning

File: SECURITY.md [hunk 1]
```diff
@@ -6,8 +6,8 @@ These versions of `@checkdigit/typescript-config` are currently being supported
 
 | Version | Supported          |
 | ------- | ------------------ |
-| \>= 3.x | :white_check_mark: |
-| \< 3.0  | :x:                |
+| \>= 5.x | :white_check_mark: |
+| \< 4.0  | :x:                |
```
Comment (le-cong) [inline SECURITY.md:10]: not sure if it should be `< 5.0` ? or should the contents in `SECURITY.md` stay the same?

**Review by le-cong — DISMISSED** [review]
lgtm, just one little thing I am not sure about. it's probably not an issue.

**Review by adcreare — APPROVED** [review]
LGTM

File: SECURITY.md (hunk 1 above)
Comment (carlansley) [reply SECURITY.md]: no it should be 5.0

**Review by adcreare — APPROVED** [review]
LGTM

**Review by ramaghanta — DISMISSED** [review]

**Review by le-cong — DISMISSED** [review]
lgtm

**Review by le-cong — DISMISSED** [review]

File: src/builder/builder.spec.mts [hunk 2]
```diff
@@ -372,6 +372,7 @@ describe('test builder', () => {
     assert.equal(output2.default.default, 'worldworld');
   });
 
+  // file.only
```
Comment (ramaghanta) [inline src/builder/builder.spec.mts:375]: delete this

**Review by ramaghanta — CHANGES_REQUESTED** [review]
one minor change

**Review by le-cong — APPROVED** [review]
lgtm

**Review by le-cong — APPROVED** [review]
lgtm

**Review by adcreare — APPROVED** [review]
LGTM

**Pull request merged**